README.md

```diff
@@ -3,49 +3,86 @@
 [![CircleCI](https://circleci.com/gh/freedomofpress/signal-protocol.svg?style=svg)](https://circleci.com/gh/freedomofpress/signal-protocol)
 [![PyPI version](https://badge.fury.io/py/signal-protocol.svg)](https://badge.fury.io/py/signal-protocol)
 
-Experimental Python bindings to [`libsignal-client`](https://github.com/signalapp/libsignal-client) Rust signal protocol implementation. This project provides a Rust extension using [PyO3](https://pyo3.rs/) to define a `signal_protocol` Python module. See [here](https://cryptography.io/en/latest/limitations.html) for a fundamental limitation storing secrets in Python-allocated memory. ⚠️USE AT YOUR OWN RISK!⚠️
+Experimental Python bindings to Rust signal protocol implementation [`libsignal-client`](https://github.com/signalapp/libsignal-client).
+This project provides a Rust extension using [PyO3](https://pyo3.rs/) to define a `signal_protocol` Python module.
+See [here](https://cryptography.io/en/latest/limitations.html) for a fundamental limitation storing secrets in Python-allocated memory.
 
-# Installation
+⚠️USE AT YOUR OWN RISK!⚠️
 
-To use the wheel distributions you do not need the Rust toolchain installed:
+## Installation
+
+To use the wheel distributions you do not need the Rust toolchain installed.
+Simply run
 
 ```
 pip install signal-protocol
 ```
 
-# Usage
+## Usage
+
+### Initial client setup
+
+The following shows how to use this library to initialize a new Signal client.
+This is the first step that must be completed before the protocol can begin.
+
+For an overview of the Signal protocol, see [this blog post](https://www.redshiftzero.com/signal-protocol/).
+Detailed [specifications](https://signal.org/docs/) are available from Signal.
 
-## Initial setup
+First, import these modules:
 
 ```py
 from signal_protocol import curve, identity_key, state, storage
+```
+
+Each client must generate a long-term identity key pair.
+This should be stored somewhere safe and persistent.
 
-# Each client must generate a long-term identity key pair.
-# This should be stored somewhere safe and persistent.
+```py
 identity_key_pair = identity_key.IdentityKeyPair.generate()
+```
 
-# Clients must generate prekeys. The example here is generating a
-# single prekey, but clients will generate many as they are one-time use
-# and consumed when a message from a new chat participant is sent. See issue #7.
+Clients must generate prekeys.
+The example generates a single prekey.
+In practice, clients will generate many prekeys, as they are one-time use and consumed when a message from a new chat participant is sent.
+(A helper function to create *n* prekeys is [forthcoming](https://github.com/freedomofpress/signal-protocol/pull/14) in a PR.)
+
+```py
 pre_key_pair = curve.KeyPair.generate()
+```
+
+Clients must generate a registration_id and store it somewhere safe and persistent.
+<!-- is this the device id? -->
```

Comment:
so registration_id and device_id are different, since one user can have multiple devices